src/tests/ftest/util/agent_utils.py

```diff
@@ -140,6 +139,24 @@ def run_agent(basepath, server_list, client_list=None):
     return sessions
 
 
+def include_local_host(hosts):
+    """Ensure the local host is included in the specified host list.
+
+    Args:
+        hosts (list): list of hosts
+
+    Returns:
+        list: list of hosts including the local host
+
+    """
+    local_host = socket.gethostname().split('.', 1)[0]
+    if hosts is None:
+        hosts = [local_host]
+    elif local_host not in hosts:
+        hosts.append(local_host)
```

Comment:
So there are situations where we need the test runner to be a client even when clients are explicitly defined?

Reply:
Currently, yes, when using the daos_api.py classes.

Reply:
Just so I am clear, even for tests where clients are defined, the test runner itself is still being a client rather than carrying out those client operations on (one of) the named client(s)?

Reply:
Previously when the tests only defined hosts using `test_machines` in their yaml file,  `run_agent()` in agent_utils.py would start an agent on the host running the test.  These changes effectively revert that behavior until we can run daos_api.py class functions from one of the clients specified in the yaml or move to using dmg (which can be configured to run from specific hosts - the test specified clients).

If a test defines clients, daos_agent will still be run on those clients, and the self.hostfile_client will only contain those hosts (so any orterun/mpirun commands using this hostfile will not include the test runner).  There will just be one additional daos_agent running on the test runner.